cmd/clusterctl/pkg/client/repository/template_test.go

```diff
@@ -0,0 +1,131 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package repository
+
+import (
+	"bytes"
+	"fmt"
+	"reflect"
+	"testing"
+
+	clusterctlv1 "sigs.k8s.io/cluster-api/cmd/clusterctl/api/v1alpha3"
+	"sigs.k8s.io/cluster-api/cmd/clusterctl/pkg/client/config"
+	"sigs.k8s.io/cluster-api/cmd/clusterctl/pkg/internal/test"
+)
+
+var templateMapYaml = []byte("apiVersion: v1\n" +
```

Comment:
nit: make this template a bit more realistic :)

Reply:
For the sake of the unit test, I'm using core objects vs Machines/Cluster etc. because it is easier to test (you don't have to deal with CRDs or schema issues), but this is ok because a template can be any yaml that complies the clusterctl contract.
(added a comment to make it clear)